### DISCUSSION TITLE
A quicker way to add Whitelisted IP ranges

### DISCUSSION THREAD

**Discussion by ProBackup-nl** [PrestaShop/PrestaShop#40196] [Ideas]
**Is your feature request related to a problem? Please describe.**
Adding IP ranges in [BO] Geolocation "Whitelisted IP addresses", like for Googlebot, Bingbot, Twitterbot, MSNBot, Applebot, etcetera is a lot of work.

**Describe the solution you'd like**
`66.249.64.0~66.249.95.255`
or
`66.249.64.0-66.249.95.255`
instead of current:
```
66.249.64
66.249.65
66.249.66
66.249.67
66.249.68
66.249.69
66.249.70
66.249.71
66.249.72
66.249.73
66.249.74
66.249.75
66.249.76
66.249.77
66.249.78
66.249.79
66.249.80
66.249.81
66.249.82
66.249.83
66.249.84
66.249.85
66.249.86
66.249.87
66.249.88
66.249.89
66.249.90
66.249.91
66.249.92
66.249.93
66.249.94
66.249.95
```
And note that this is only for 1 range. I need to describe [30+ search engine bot ranges with ips](https://www.techarp.com/guides/search-engine-robot-ip-addresses/).

**Describe alternatives you've considered**
There are no alternatives.

**Comment by khouloudbelguith** [comment]
Hi @ProBackup-nl,

Thanks for your report.
Ping @colinegin, @marionf what do you think?
Is it possible to replace the Whitelisted IP addresses to IP ranges?
![image](https://user-images.githubusercontent.com/16067358/60888237-3b723500-a24e-11e9-8719-cb58310feb34.png)

Thanks!

**Comment by MathiasReker** [comment]
@ProBackup-nl  why would you want to whitelist all those IPs?

**Comment by ProBackup-nl** [comment]
@MathiasReker I want the webshop only available for the Netherlands (Geolocation), the scope is even more local: 30 zip codes out of 4000 Dutch 4-digit postal codes. Thus my offering is very local. Though still want to be found in Google and other locally used search engines (not Yandex or Baidu).

**Comment by colinegin** [comment]
What do you think about it @PierreRambaud ?

**Comment by PierreRambaud** [comment]
Totally agreed to allow ip ranges. :+1:

**Comment by Hlavtox** [comment]
You can make your own script or something. Converting to idea.
